### PR TITLE
New definition for 'react-virtual-keyboard'

### PR DESCRIPTION
**Pull request by bsurai**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not provide its own types, and you can not add them.
- [x] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] `tslint.json` should be present, and `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, and `strictNullChecks` set to `true`.

If changing an existing definition:
- [ ] Provide a URL to documentation or source code which provides context for the suggested changes: <<url here>>
- [ ] Increase the version number in the header if appropriate.
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

If removing a declaration:
- [ ] If a package was never on DefinitelyTyped, you don't need to do anything. (If you wrote a package and provided types, you don't need to register it with us.)
- [ ] Delete the package's directory.
- [ ] Add it to `notNeededPackages.json`.

### CODE REVIEW

**Comment by bsurai** [comment]
@tdreyno @dt-bot @leemhenson @farcaller @altano  This definition does not have these dependencies. Why I've got these errors?

```
npm ERR! extraneous: @types/charm@1.0.0 /home/travis/build/DefinitelyTyped/DefinitelyTyped/node_modules/types-publisher/node_modules/@types/charm
npm ERR! extraneous: @types/fs-extra@3.0.3 /home/travis/build/DefinitelyTyped/DefinitelyTyped/node_modules/types-publisher/node_modules/@types/fs-extra
npm ERR! extraneous: @types/mz@0.0.31 /home/travis/build/DefinitelyTyped/DefinitelyTyped/node_modules/types-publisher/node_modules/@types/mz
npm ERR! extraneous: @types/node-fetch@1.6.7 /home/travis/build/DefinitelyTyped/DefinitelyTyped/node_modules/types-publisher/node_modules/@types/node-fetch
npm ERR! extraneous: @types/oboe@2.0.28 /home/travis/build/DefinitelyTyped/DefinitelyTyped/node_modules/types-publisher/node_modules/@types/oboe
npm ERR! extraneous: @types/semver@5.3.31 /home/travis/build/DefinitelyTyped/DefinitelyTyped/node_modules/types-publisher/node_modules/@types/semver
npm ERR! extraneous: @types/source-map-support@0.2.28 /home/travis/build/DefinitelyTyped/DefinitelyTyped/node_modules/types-publisher/node_modules/@types/source-map-support
npm ERR! extraneous: @types/tar@1.0.29 /home/travis/build/DefinitelyTyped/DefinitelyTyped/node_modules/types-publisher/node_modules/@types/tar
npm ERR! extraneous: @types/yargs@6.6.0 /home/travis/build/DefinitelyTyped/DefinitelyTyped/node_modules/types-publisher/node_modules/@types/yargs
```

File: types/react-virtual-keyboard/tsconfig.json [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+        "react-virtual-keyboard-test.ts"
+    ],
+    "compilerOptions": {
+        "module": "commonjs",
+        "lib": [
+            "es6",
+            "dom"
+        ],
+        "noImplicitAny": true,
+        "noImplicitThis": false,
```
Comment (andy-ms) [inline types/react-virtual-keyboard/tsconfig.json]: This must be `true` for new packages.

File: types/react-virtual-keyboard/index.d.ts [hunk 2]
```diff
@@ -0,0 +1,133 @@
+// Type definitions for react-virtual-keyboard 1.0.0
+// Project: https://www.npmjs.com/package/react-virtual-keyboard
+// Definitions by: Bogdan Surai <https://github.com/bsurai>
+//                 
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+// TypeScript Version: 2.4
+
+declare module "virtual-keyboard" {
+    import * as jQuery from "jquery";
+    // import "virtual-keyboard/dist/js/jquery.keyboard.js";
```
Comment (andy-ms) [inline types/react-virtual-keyboard/index.d.ts]: These comments should be removed.

File: types/react-virtual-keyboard/index.d.ts [hunk 3]
```diff
@@ -0,0 +1,133 @@
+// Type definitions for react-virtual-keyboard 1.0.0
+// Project: https://www.npmjs.com/package/react-virtual-keyboard
+// Definitions by: Bogdan Surai <https://github.com/bsurai>
+//                 
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+// TypeScript Version: 2.4
+
+declare module "virtual-keyboard" {
```
Comment (andy-ms) [inline types/react-virtual-keyboard/index.d.ts]: You should create a `types/virtual-keyboard` package for this.

**Review by ghost — CHANGES_REQUESTED** [review]
Needs a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

**Comment by andy-ms** [comment]
@bsurai Those look like `types-publisher`'s devDependencies. Not sure why NPM doesn't like them.

File: types/react-virtual-keyboard/index.d.ts [hunk 4]
```diff
@@ -0,0 +1,133 @@
+// Type definitions for react-virtual-keyboard 1.0.0
+// Project: https://www.npmjs.com/package/react-virtual-keyboard
+// Definitions by: Bogdan Surai <https://github.com/bsurai>
+//
```
Comment (andy-ms) [inline types/react-virtual-keyboard/index.d.ts]: Remove this blank line.

**Comment by andy-ms** [comment]
Please change this to create two packages, `types/virtual-keyboard` and `types/react-virtual-keyboard`, instead of defining them to gether in `types/react-virtual-keyboard`.

**Comment by bsurai** [comment]
@andy-ms. Here is ready definition for 'virtual-keyboard'
https://github.com/DefinitelyTyped/DefinitelyTyped/pull/18441

**Comment by bsurai** [comment]
If 'virtual-keyboard' is OK, I will add 'react-virtual-keyboard'

**Comment by mhegazy** [comment]
looks like this is stale now. please reopen if still needed.

**Pull request closed without merging**